File: test-testcontainers/src/test/java/io/micronaut/test/junit5/resource/TestResourceTest.java
package io.micronaut.test.junit5.resource;

import java.net.URI;

import io.micronaut.context.annotation.Property;
import io.micronaut.core.io.socket.SocketUtils;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest
public class TestResourceTest {

    @Test
    void testResourceAvailable(@Property(name = "mongodb.uri") URI uri) {
        final int port = uri.getPort();
        Assertions.assertFalse(SocketUtils.isTcpPortAvailable(port));
    }
}

File: test-kotest/src/test/kotlin/io/micronaut/test/kotest/TestResourceTest.kt
package io.micronaut.test.kotest

import io.kotest.core.spec.style.BehaviorSpec
import io.kotest.matchers.shouldBe
import io.micronaut.context.annotation.Property
import io.micronaut.core.io.socket.SocketUtils
import io.micronaut.test.extensions.kotest.annotation.MicronautTest
import java.net.URI

@MicronautTest
class TestResourceTest(
    @Property(name="mongodb.uri") uri : URI
) : BehaviorSpec ({
    given("the a test resource is available service") {

        `when`("the port is checked for availability") {
            val tcpPortAvailable = SocketUtils.isTcpPortAvailable(uri.port)
            then("the service should be runnning") {
                tcpPortAvailable shouldBe false
            }
        }
    }
})